src/Search/Output/Spreadsheet.php

```diff
@@ -0,0 +1,544 @@
+            $name = $val["name"];
+
+            // prefix by group name (corresponding to optgroup in dropdown) if exists
+            if (isset($val['groupname'])) {
+                $groupname = $val['groupname'];
+                if (is_array($groupname)) {
+                    $groupname = $groupname['name'];
+                }
+                $name  = "$groupname - $name";
+            }
+
+            // Not main itemtype add itemtype to display
+            if ($data['itemtype'] != $val['itemtype']) {
+                if (!isset($metanames[$val['itemtype']])) {
+                    if ($metaitem = getItemForItemtype($val['itemtype'])) {
+                        $metanames[$val['itemtype']] = $metaitem->getTypeName();
+                    }
+                }
+                $name = sprintf(
+                    __('%1$s - %2$s'),
+                    $metanames[$val['itemtype']],
+                    $val["name"]
+                );
+            }
+
+            $worksheet->setCellValue([$col_num, $line_num], $name);
+        }
+
+        // Add specific column Header
+        if (isset($CFG_GLPI["union_search_type"][$data['itemtype']])) {
+            ++$col_num;
+            $worksheet->setCellValue([$col_num, $line_num], __('Item type'));
+        }
+
+        $typenames = [];
+        // Display Loop
+        foreach ($data['data']['rows'] as $row) {
+            $col_num = 0;
+            ++$line_num;
+
+            // Print other toview items
+            foreach ($data['data']['cols'] as $col) {
+                ++$col_num;
+                $colkey = "{$col['itemtype']}_{$col['id']}";
+
+                $value = DataExport::normalizeValueForTextExport($row[$colkey]['displayname'] ?? '');
+                $worksheet->setCellValue([$col_num, $line_num], $value);
+            }
+
+            if (isset($CFG_GLPI["union_search_type"][$data['itemtype']])) {
+                ++$col_num;
+                if (!isset($typenames[$row["TYPE"]])) {
+                    if ($itemtmp = getItemForItemtype($row["TYPE"])) {
+                        $typenames[$row["TYPE"]] = $itemtmp->getTypeName();
+                    }
+                }
+                $value = DataExport::normalizeValueForTextExport($typenames[$row["TYPE"]] ?? '');
+                $worksheet->setCellValue([$col_num, $line_num], $value);
+            }
+        }
+
+        header('Content-Type: ' . $output->getMime());
+        header('Content-Disposition: attachment; filename="' . urlencode($output->getFileName()) . '"');
+        $writer->save('php://output');
+    }
+
+    public function getWriter(): BaseWriter
+    {
+        return $this->writer;
+    }
+
+    public function getSpreasheet(): \PhpOffice\PhpSpreadsheet\Spreadsheet
+    {
+        return $this->spread;
+    }
+
+    /**
+     * Get file title
+     *
+     * @param array $data Array data of search
+     *
+     * @return string Title
+     **/
+    protected function getTitle(array $data): string
```

Comment:
I may miss something, but when I open the spreadsheet in LibreOffice, I do not see where the title is displayed.

If it shares the same code as `PDFSearchOutput`, maybe this whole can can be moved into the `ExportSearchOutput::computeTitle()` method. As far as I understand the code, it will not have any side effects on other export methods (that are only `NamesListSearchOutput`).

Reply:
For new methods, title is set in metadata properties:
![image](https://github.com/glpi-project/glpi/assets/224733/2cd53a3b-0dac-4c4e-8368-7861cf18973d)

Since we now use phpSpreadsheet for all exports but PDF; yes I think we can move that part on parent class.

Reply:
> Best option would be to use phpspreadsheet to export PDF aswell; and then we can drop all old stuff.

Indeed, we could use this lib to generate PDFs.

Reply:
The rendering for pdf was bad, no ?

Reply:
Yes, even if it can be changed.
Trying to factorize old echoes with new object oriented lib will be difficult. Technically speaking, the best way is to use new way everywhere; until that we can just keep legacy code.

Reply:
I'll take an eye at PDF exports; I did not git into that yet.

Reply:
After a bit of work, I have something that's already better than the original:
![image](https://github.com/glpi-project/glpi/assets/224733/3b8987d1-216d-412e-8bd1-3920028ac285)

There is still work; it's not very difficult, but lack of documentation on PDF output specificities make some changes complicated.